server/server.go

```diff
@@ -189,23 +203,79 @@ func (s *audioHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *stickerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	messageIDs, ok := r.URL.Query()["messageId"]
+	if !ok || len(messageIDs) == 0 {
+		s.logger.Error("no messageID")
+		return
+	}
+	messageID := messageIDs[0]
+	var stickerHash string
+	err := s.db.QueryRow(`SELECT sticker_hash FROM user_messages WHERE id = ?`, messageID).Scan(&stickerHash)
+	if err != nil {
+		s.logger.Error("failed to find sticker", zap.Error(err))
+		return
+	}
+	if len(stickerHash) == 0 {
+		s.logger.Error("invalid sticker hash")
+		return
+	}
+
+	content, err := s.downloader.Get(stickerHash)
+	if err != nil {
+		s.logger.Error("could not download hash", zap.Error(err))
+		return
+	}
+
+	w.Header().Set("Cache-Control", "max-age:290304000, public")
+	w.Header().Set("Expires", time.Now().AddDate(60, 0, 0).Format(http.TimeFormat))
```

Comment:
60 years ?

Reply:
yes. In practice it means that the resource will never expire, and kept in browser cache